### ISSUE TITLE
[Documentation] Security_groups parameter in aws_instance

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#255]
_This issue was originally opened by @fjromerom as hashicorp/terraform#8714. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

aws_instance from documentation:

> security_groups - (Optional) A list of security group names to associate with. If you are creating Instances in a VPC, use vpc_security_group_ids instead.

security_groups parameter does not work with names, It works with IDs.

If I set security groups names terraform prints:

`* aws_instance.bastion: Error launching source instance: InvalidParameterCombination: The parameter groupName cannot be used with the parameter subnet
    status code: 400, request id: XXXXXXXXXXXXXXXXXXX`

**Comment by n3ph** [comment]
Can't reproduce...

```console
$ terraform --version
Terraform v0.11.7
+ provider.aws v1.25.0
```

```tf
#------------------------------------------------------------------------------#
# Test Setup
#------------------------------------------------------------------------------#

provider "aws" {
  max_retries = 3
  region      = "eu-central-1"
  profile     = "devops"
}

data "aws_vpc" "default" {
  default = true
}

data "aws_security_group" "default" {
  vpc_id = "${data.aws_vpc.default.id}"
}

#------------------------------------------------------------------------------#
# Amazon2 Base AMI
#------------------------------------------------------------------------------#

data "aws_ami" "amzn2" {
  most_recent = true

  filter {
    name   = "name"
    values = ["amzn2-ami-hvm*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  owners = ["159388291991"]
}

#------------------------------------------------------------------------------#
# Instance
#------------------------------------------------------------------------------#

resource "aws_instance" "web" {
  ami             = "${data.aws_ami.amzn2.id}"
  instance_type   = "t2.micro"
  security_groups = ["${data.aws_security_group.default.name}"]
}
```

```console
$ terraform apply
data.aws_vpc.default: Refreshing state...
data.aws_ami.amzn2: Refreshing state...
data.aws_subnet_ids.default: Refreshing state...
data.aws_security_group.default: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + aws_instance.web
      id:                           <computed>
      ami:                          "ami-09a2e326dc3bfcc2a"
      associate_public_ip_address:  <computed>
      availability_zone:            <computed>
      ebs_block_device.#:           <computed>
      ephemeral_block_device.#:     <computed>
      get_password_data:            "false"
      instance_state:               <computed>
      instance_type:                "t2.micro"
      ipv6_address_count:           <computed>
      ipv6_addresses.#:             <computed>
      key_name:                     <computed>
      network_interface.#:          <computed>
      network_interface_id:         <computed>
      password_data:                <computed>
      placement_group:              <computed>
      primary_network_interface_id: <computed>
      private_dns:                  <computed>
      private_ip:                   <computed>
      public_dns:                   <computed>
      public_ip:                    <computed>
      root_block_device.#:          <computed>
      security_groups.#:            "1"
      security_groups.3814588639:   "default"
      source_dest_check:            "true"
      subnet_id:                    <computed>
      tenancy:                      <computed>
      volume_tags.%:                <computed>
      vpc_security_group_ids.#:     <computed>


Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_instance.web: Creating...
  ami:                          "" => "ami-09a2e326dc3bfcc2a"
  associate_public_ip_address:  "" => "<computed>"
  availability_zone:            "" => "<computed>"
  ebs_block_device.#:           "" => "<computed>"
  ephemeral_block_device.#:     "" => "<computed>"
  get_password_data:            "" => "false"
  instance_state:               "" => "<computed>"
  instance_type:                "" => "t2.micro"
  ipv6_address_count:           "" => "<computed>"
  ipv6_addresses.#:             "" => "<computed>"
  key_name:                     "" => "<computed>"
  network_interface.#:          "" => "<computed>"
  network_interface_id:         "" => "<computed>"
  password_data:                "" => "<computed>"
  placement_group:              "" => "<computed>"
  primary_network_interface_id: "" => "<computed>"
  private_dns:                  "" => "<computed>"
  private_ip:                   "" => "<computed>"
  public_dns:                   "" => "<computed>"
  public_ip:                    "" => "<computed>"
  root_block_device.#:          "" => "<computed>"
  security_groups.#:            "" => "1"
  security_groups.3814588639:   "" => "default"
  source_dest_check:            "" => "true"
  subnet_id:                    "" => "<computed>"
  tenancy:                      "" => "<computed>"
  volume_tags.%:                "" => "<computed>"
  vpc_security_group_ids.#:     "" => "<computed>"
aws_instance.web: Still creating... (10s elapsed)
aws_instance.web: Still creating... (20s elapsed)
aws_instance.web: Creation complete after 23s (ID: i-064ca20c55b301830)
```

**Comment by ryndaniels** [comment]
Looks like this was fixed a while back - looking at #5680 and seeing that the documentation now specifies when to use IDs vs names. 👍
